test/dict.jl

```diff
@@ -1039,13 +1039,35 @@ end
 end
 
 @testset "copy!" begin
-    s = Dict(1=>2, 2=>3)
-    for a = ([3=>4], [0x3=>0x4], [3=>4, 5=>6, 7=>8], Pair{UInt,UInt}[3=>4, 5=>6, 7=>8])
-        @test s === copy!(s, Dict(a)) == Dict(a)
-        if length(a) == 1 # current limitation of Base.ImmutableDict
-            @test s === copy!(s, Base.ImmutableDict(a[])) == Dict(a[])
+    @testset "copy!(::$(typeof(s)), _)" for s in Any[
+        Dict(1 => 2, 2 => 3),                      # concrete key type
+        Dict{Union{Int,UInt},Int}(1 => 2, 2 => 3), # union key type
+        Dict{Any,Int}(1 => 2, 2 => 3),             # abstract key type
+        Dict{Int,Float64}(1 => 2, 2 => 3),         # values are converted
+    ]
```

Comment:
Style nit: indent for `for` and indent for `[` should make at least 2:
```suggestion
            Dict(1 => 2, 2 => 3),                      # concrete key type
            Dict{Union{Int,UInt},Int}(1 => 2, 2 => 3), # union key type
            Dict{Any,Int}(1 => 2, 2 => 3),             # abstract key type
            Dict{Int,Float64}(1 => 2, 2 => 3),         # values are converted
        ]
```